class.eventcalendar.plugin.php

```diff
@@ -199,28 +200,33 @@ public function postController_beforeBodyInput_handler($sender) {
         $sender->addJsFile('eventcalendar.js', 'plugins/EventCalendar');
         $sender->addDefinition(
             'EventCalendarCategoryIDs',
-            json_encode(c('EventCalendar.CategoryIDs'))
+            json_encode(unserialize(c('EventCalendar.CategoryIDs')))
         );
 
-        $categoryID = $sender->Discussion->CategoryID;
-
+        $categoryID = $sender->Data['Category']['CategoryID'];
         // initially don't hide elements in allowed categories
         $cssClass = 'P EventCalendarInput';
-        if (!in_array($categoryID, c('EventCalendar.CategoryIDs'))) {
-            $cssClass .= ' Hidden';
+        if (!in_array($categoryID, unserialize(c('EventCalendar.CategoryIDs')))) {
+            $cssClass .= ' hidden';
         }
-
         $year = date('Y');
         $yearRange = $year.'-'.($year + 1);
-        $fields = explode(',', t('EventCalendar.DateOrder', 'month,day,year'));
+        $fields = explode(',', t('EventCalendar.DateOrder', 'day,month,year'));
```

Comment:
As a German I also prefer that format, but the other one is - as far as I know - more common in English language and just to be consistent, I have chosen it here and would like to stick with it. It's customizable by using the language file

Reply:
Yes, but i'm not used to develop using Vanilla features and as it was just for one site .. So i made it quick and dirty !

As the title says.. If it can help, it's cool, if not, sorry !